### DISCUSSION TITLE
Combine Query and Mutation cache

### DISCUSSION THREAD

**Discussion by danactive** [TanStack/query#1422] [General] [answered]
v3 add a Mutation Cache. In my use case I want my queries and mutations united into one place so I can get the latest data. Is the best practice to define an onSuccess of mutation to set query data?

```
onSuccess: (result) => {
      queryClient.setQueryData('todos', result)
}
```

Then to extract the data

`queryClient.getQueryData('todos');`

This way I don't need to deal with the MutationCache

**Comment by TkDodo** [comment, marked as answer]
Yes. You only have to deal with queryClient, but it works the same as with queryCache before.
